pkg/cmd/cli/cmd/newapp.go

```diff
@@ -446,6 +446,10 @@ func setupAppConfig(f *clientcmd.Factory, out io.Writer, c *cobra.Command, args
 	if config.AllowMissingImages && config.AsSearch {
 		return cmdutil.UsageError(c, "--allow-missing-images and --search are mutually exclusive.")
 	}
+
+	if len(config.SourceImage) != 0 && len(config.SourceImagePath) == 0 {
+		return cmdutil.UsageError(c, "--source-image-path must be specified when --source-image is specified.")
+	}
```

Comment:
Shouldn't we check the opposite as well, `--source-image-path` is set while `--source-image` is empty?